clang/lib/Sema/Metafunctions.cpp

```diff
@@ -1006,6 +1011,37 @@ bool isAccessible(Sema &S, DeclContext *AccessDC, NamedDecl *D) {
   return Result;
 }
 
+template <typename T>
```

Comment:
Does this need to be a template? Seems like `T` will always be `CXXMethodDecl`.

Reply:
Ahhh I see, sorry about that. How about we just take the `ExceptionSpecificationType`, and inline this as a lambda in the `isFunctionOrLambdaNoexcept` function?

```cpp
static bool isFunctionOrLambdaNoexcept(const QualType QT) {
  static auto isExceptionSpecNoexcept = [](ExceptionSpecificationType) {
    switch (...) // ...
  }
  
  if (auto *FPT = dyn_cast<FunctionProtoType>(QT)) {
    return isExceptionSpecNoexcept(FPT->getExceptionSpecType());
  } else if (auto *RT = dyn_cast<RecordType>(QT)) {
    auto *RDecl = cast<CXXRecordDecl>(RT->getDecl());
    if (RDecl->isLambda() && !RDecl->isGenericLambda())
      return isExceptionSpecNoexcept(
          RDecl->getLambdaCallOperator()->getExceptionSpecType());
  }
  return false;
}
```

What do you think?

Reply:
Sure, done

Reply:
not relevant for the latest version - the logic was removed